### PR TITLE
Refactor adapter to move FileSettings from XMLAdapter to XMLConfig

### LINKED ISSUE #592: Refactor adapter to use action config instead of file settings.
#### Broken rules:

In order to be in line with other toolkit adapters, the adapter needs to be refactored to move the file settings from a private attribute in the `XMLAdapter` to the XML action configs instead. 

#### Suggestions to restore compliance:

- [x] Remove file settings attribute from `XMLAdapter`.
- [x] Add file settings attribute to `XMLConfig`.
- [x] Create new interface `IXMLConfig` to force file settings on all xml configs
- [x] Update CRUD methods to use the refactored attribute.

### PR DESCRIPTION
**Pull request by Tom-Kingstone**
### Issues addressed by this PR

 Closes #592 

 
- File settings attribute removed from `XMLAdapter` and placed in `XMLConfig`.
- `XMLConfig` is now an `IXMLConfig` and any future configs must contain a file settings attribute.
- CRUD methods have been updated to use `config.File` rather than `_fileSettings`

### CODE REVIEW

**Comment by Tom-Kingstone** [comment]
@BHoMBot check compliance

File: XML_Adapter/XMLAdapter.cs [hunk 1]
```diff
@@ -43,25 +43,9 @@ public partial class XMLAdapter : BHoMAdapter
         [Description("Specify XML file and properties for data transfer")]
         [Input("fileSettings", "Input the file settings to get the file name and directory the XML Adapter should use")]
         [Output("adapter", "Adapter to XML")]
-        public XMLAdapter(BH.oM.Adapter.FileSettings fileSettings = null)
+        public XMLAdapter()
```
Comment (FraserGreenroyd) [inline XML_Adapter/XMLAdapter.cs:46]: Needs versioning please

**Review by FraserGreenroyd — CHANGES_REQUESTED** [review]
Looks like I set up the settings nicely the last time so this wasn't as much work as I was expecting!

Minor changes needed please @Tom-Kingstone

**Comment by Tom-Kingstone** [comment]
@BHoMBot check compliance
@BHoMBot check versioning

File: XML_Adapter/XMLAdapter.cs [hunk 2]
```diff
@@ -40,28 +40,12 @@ namespace BH.Adapter.XML
 {
     public partial class XMLAdapter : BHoMAdapter
     {
-        [Description("Specify XML file and properties for data transfer")]
-        [Input("fileSettings", "Input the file settings to get the file name and directory the XML Adapter should use")]
-        [Output("adapter", "Adapter to XML")]
-        public XMLAdapter(BH.oM.Adapter.FileSettings fileSettings = null)
+        [PreviousVersion("7.0", "BH.Adapter.XML.XMLAdapter(BH.oM.Adapter.Filesettings)")]
```
Comment (FraserGreenroyd) [inline XML_Adapter/XMLAdapter.cs:43]:
```suggestion
        [PreviousVersion("7.0", "BH.Adapter.XML.XMLAdapter(BH.oM.Adapter.FileSettings)")]
```

File: XML_Adapter/XMLAdapter.cs [hunk 3]
```diff
@@ -40,28 +40,12 @@ namespace BH.Adapter.XML
...
+        [Description("Specify XML file and properties for data transfer.")]
```
Comment (FraserGreenroyd) [inline XML_Adapter/XMLAdapter.cs:44]: I think this needs updating to something like `Connect to XML interoperability within the BHoM.`

**Review by FraserGreenroyd — CHANGES_REQUESTED** [review]

**Comment by Tom-Kingstone** [comment]
@BHoMBot check required
@BHoMBot check compliance

**Review by FraserGreenroyd — APPROVED** [review]
Looks good and aligns  nicely @Tom-Kingstone - great work

**Comment by FraserGreenroyd** [comment]
@BHoMBot this is a DevOps instruction. I am authorising dispensation to be granted on check ref. 18073278039

**Comment by FraserGreenroyd** [comment]
@BHoMBot check ready-to-merge

**Pull request merged**